x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/NativeAutodetectProcessFactory.java

```diff
@@ -71,26 +71,27 @@ void setProcessConnectTimeout(TimeValue processConnectTimeout) {
     }
 
     @Override
-    public AutodetectProcess createAutodetectProcess(Job job,
+    public AutodetectProcess createAutodetectProcess(String pipelineId,
+                                                     Job job,
```

Comment:
One of the requirements was that this upgrade be possible WHILE the referenced job is running. Consequently, the snapshot upgrade task and the job task COULD be assigned to the same node. If the pipeline ID was not given directly, this would cause a file name conflict.

Admittedly, there is already this "unique pipeline flag" but that is a `long` value. I thought it would be nice to include the snapshot ID directly in the pipeline name. It makes the resulting logs very easy to investigate snapshot upgrader issues by looking for `<job_id>-<snapshot_id>`